inference-engine/src/vpu/graph_transformer/src/stages/pad.cpp

```diff
@@ -93,20 +93,58 @@ void FrontEnd::parsePad(const Model& model, const ie::CNNLayerPtr& _layer, const
     auto layer = std::dynamic_pointer_cast<ie::PadLayer>(_layer);
     IE_ASSERT(layer != nullptr);
 
-    IE_ASSERT(layer->pads_begin.size() == 4);
-    IE_ASSERT(layer->pads_end.size() == 4);
+    const auto ndims = inputs[0]->desc().dimsOrder().numDims();
+    VPU_THROW_UNLESS(ndims == 3 || ndims == 4, "Layer %s support only 3D and 4D input", layer->name);
```

Comment:
When you will rebase your branch, please, mention the actual ndims which is provided

Reply:
fixed